ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/TransactionsReader.scala

```diff
@@ -62,41 +62,53 @@ private[dao] final class TransactionsReader(
       filter: FilterRelation,
       verbose: Boolean,
   ): Source[(Offset, GetTransactionsResponse), NotUsed] = {
-    val events: Source[EventsTable.Entry[Event], NotUsed] =
-      PaginatingAsyncStream.streamFrom((startExclusive, Option.empty[Int]), getOffset[Event]) {
-        case (prevOffset, prevNodeIndex) =>
-          val query =
-            EventsTable
-              .preparePagedGetFlatTransactions(sqlFunctions)(
-                startExclusive = prevOffset,
-                endInclusive = endInclusive,
-                filter = filter,
-                pageSize = pageSize,
-                previousEventNodeIndex = prevNodeIndex,
-              )
-              .withFetchSize(Some(pageSize))
-          val rawEventsFuture =
-            dispatcher.executeSql(dbMetrics.getFlatTransactions) { implicit connection =>
-              query.asVectorOf(EventsTable.rawFlatEventParser)
-            }
-          rawEventsFuture.flatMap(
-            rawEvents =>
-              Timed.future(
-                future = Future.traverse(rawEvents)(deserializeEntry(verbose)),
-                timer = dbMetrics.getFlatTransactions.translationTimer,
+
+    val eventsRangeF: Future[EventsRange[Long]] = dispatcher.executeSql(dbMetrics.getRowIdRange) {
+      connection =>
+        EventsRange.eventsRange(EventsRange(startExclusive, endInclusive))(connection)
+    }
```

Comment:
Would it make sense to shoot the two queries in a single transaction? The round-trip could be detrimental.

Reply:
Mmmmh, looks like it's not feasible. Not easily, at least.

Reply:
I was thinking about this, did not find an easy way to fuse this query into the same transaction context as the pagination query... but in any case we only need to run this query once, before we get the first page, after this we use `row_id` for pagination.

Reply:
BTW @S11001001 implemented `SqlSequence` which we can use: https://github.com/digital-asset/daml/blob/1adc69efadac2f1e34547b1d66faa260f2b68f1e/ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/SqlSequence.scala